### ISSUE TITLE
SSL Error with `networkGet('https://...')`

### ISSUE THREAD

**Issue by hluk** [hluk/CopyQ#1177]
Following could throw an error `Error creating SSL context ()`.

```js
url = 'https://raw.githubusercontent.com/hluk/copyq-commands/master/Automatic/import-commands-after-copied.ini'
reply = networkGet(url)
var commandsData = str(reply.data)
if (reply.status != 200) {
    throw 'Failed to fetch commands.'
        + '\\nStatus code: ' + reply.status
        + '\\nError: ' + reply.error
}
```

Looks like a problem with missing SSL libraries on Windows: https://stackoverflow.com/questions/18894056/error-creating-ssl-context-qt

**Comment by crazytriangle333** [comment]
Hi, I get a similar error when calling `networkGet()` on Window 10. This is returned in the `reply.error` when downloading e. g. from Github over `https://`:

`TLS initialization failed`

I think it can be related. Could it be that I need to install something to my OS, or could it be that CopyQ is only using an older Qt / SSL library?

**Comment by pbodnar** [comment]
Hi, I get a similar error when calling `networkGet()` on Window 10. This is returned in the `reply.error` when downloading e. g. from Github over `https://`:

`TLS initialization failed`

I think it can be related. Could it be that I need to install something to my OS, or could it be that CopyQ is only using an older Qt / SSL library? I use the latest CopyQ v3.10.0.

**Comment by hluk** [comment]
Can you test it with [this build](https://ci.appveyor.com/project/hluk/copyq/builds/33393089/job/a7g76s9be89gs1ip/artifacts)? I've included some OpenSSL libraries which might help.

**Comment by pbodnar** [comment]
> Can you test it with [this build](https://ci.appveyor.com/project/hluk/copyq/builds/33393089/job/a7g76s9be89gs1ip/artifacts)? I've included some OpenSSL libraries which might help.

Hi, it still returns the same error. I tried just the portable version (from the link) after quitting the installed version, I hope that should be sufficient for a test...

**Comment by hluk** [comment]
Can you test with [yet another build](https://ci.appveyor.com/project/hluk/copyq/builds/33572801/job/jm9llr5m1myvly2q/artifacts)? I've included different versions of the libraries.

The problem is that I don't have a way to test it on the build server and I don't see an easy way to find out which exact versions of libraries are needed so this can break easily when Qt is updated.

**Comment by pbodnar** [comment]
Congrats, it works this time! :) So it seems like exchanging the dll libraries did the trick...

**Comment by hluk** [comment]
> Congrats, it works this time! :) So it seems like exchanging the dll libraries did the trick...

Great!

I still haven't figured out how to test it. Without the test it will likely break in future releases.

**Comment by pbodnar** [comment]
> I still haven't figured out how to test it. Without the test it will likely break in future releases.

You probably mean automated testing on possibly different versions of Windows on the build server directly, right? I'm far from being an expert on this, but it looks like there are some paid services for running virtual machines (like from Amazon), or of course you can setup and run VMs on your PC directly (How-To Geek contains pretty good articles like [this](https://www.howtogeek.com/197559/how-to-install-windows-10-on-your-pc/) or [this](https://www.howtogeek.com/206286/stop-testing-software-on-your-pc-use-virtual-machine-snapshots-instead/)) + possibly use something like [Vagrant](https://en.wikipedia.org/wiki/Vagrant_(software)) to automate the VMs' setup and running. Either way, it surely takes some time to find the way...

**Comment by hluk** [comment]
@pbodnar Thanks for suggestions.

I gave up running Windows in VM (at least for now) because it breaks too often - recently, I couldn't run the VM image on a new machine.

But I wasn't aware that Vagrant could spin up a Windows VM -- this would be awesome.

I'm using [Appveyor](appveyor.com) to build the binary for Windows and run the tests.

Anyway, I managed to make the `networkGet()` tests fail by removing some installed SSL dlls on the VM and clearing PATH env variable. Hopefully, this catches most of the problems with missing SSL dlls.
